web3.go

```diff
@@ -70,6 +71,30 @@ func WeiAsGwei(w *big.Int) string {
 	return new(big.Rat).SetFrac(w, weiPerGwei).FloatString(9)
 }
 
+// IntAsFloat converts a *big.Int (ie: wei), to *big.Float (ie: ETH)
+func IntAsFloat(i *big.Int, decimals int) *big.Float {
```

Comment:
This loses precision, but you can use a generalized form of `WeiAsBase` instead: https://play.golang.org/p/JCbpP9RgS4Z

Reply:
This doesn't lose precision, it's your ToString() that drops it, see this: https://play.golang.org/p/beXqCL_bff7

Reply:
It's pretty unintuitive that calling `f.String()` behaves differently from `fmt.Print(f)`. How about returning string from this func so the caller can't screw it up?

Reply:
But then you don't get the float, this is also a helper library too and returning a string here is only useful for the CLI tool so that's the only place it should be probably. You can see it sets the precision in the docs: https://golang.org/pkg/math/big/#Float.String

Reply:
I see that `String()` sets the precision, but it's not obvious to me why the default formatter uses a _different_ precision. I don't know what else users would do with the float other than format it as a string. We should encourage them to do any necessary manipulation of the value on the `big.Int` ahead of time, so they don't screw up the float.

Reply:
If you're working with floats, like this is, why wouldn't you work with floats?  The whole point of big floats is that you can work with them without losing precision.

Reply:
I'm suggesting that we should not encourage working with floats because it is inherently risky, e.g. you can easily produce values which cannot be represented in the original format: https://play.golang.org/p/Nz5GAKRGIo9
In this case it is _too much_ precision that is causing the problem, namely producing a value which is half of a wei.

Reply:
Here are more dramatic cases: 
https://play.golang.org/p/KASKVWfZnJN
https://play.golang.org/p/60SmyWw8tFy